File: nntest.ts
/**
 * test cpp interface
 */
namespace serial {
    /**
     * Set serial receive buffer size.
     * @param size the new size of the receive buffer, eg: 20
     */
    //% blockId=serial_buffersize block="serial receive buffer size %size"
    //% shim=serial::setReceiveBufferSize
    export function setReceiveBufferSize(size: number): void {
        return;
    }

    /**
     * Reset serial back to USBTX/USBRX.
     */
    //% blockId=serial_resetserial block="serial reset"
    //% shim=serial::resetSerial
    export function resetSerial(): void {
        return;
    }

    //% shim=serial::read
    export function read(delimiters: string): string {
        return "OK";
    }

}

File: nntest.cpp
#include "pxt.h"

namespace serial {
    //%
    void setReceiveBufferSize(int size) {
    	uBit.serial.send("size: ");
    	for (int i=0; i<size; i++) {
        	uBit.serial.send(".");
    	}
    	uBit.serial.send("\r\n");
    }

    //%
    void resetSerial() {
    	uBit.serial.send("resetSerial 0.1\r\n");
    }

    //%
    StringData *read(StringData *delimiters) {
    	const char *jsonNN = PXT_STRING_DATA(json);
    	uBit.serial.send("read: ");
    	uBit.serial.send(jsonNN);
    	uBit.serial.send("\r\n");
    	ManagedString result = ManagedString("Ergebnis");
        return result.leakData();
    }
}
